hls4ml/model/optimizer/passes/multi_dense.py

```diff
@@ -5,7 +5,7 @@
 class ReplaceMultidimensionalDenseWithConv(OptimizerPass):
     def match(self, node):
         return isinstance(node, Dense) and \
-            len(node.get_input_variable().shape) > 1
+            len(node.get_input_variable().shape) - sum(d==1 for d in node.get_input_variable().shape) > 1
```

Comment:
Surely there is a more descriptive way of excluding RNN layers from this optimization... :smiley:

Reply:
Indeed :) Although it seemed to me like a multidimensional dense layer with all axis lengths but one equal to 1 should not really be automatically transformed into a pointwise conv. I can change this to explicitly pick out the dense layers connected to LSTM/GRU if you would prefer.